internal/testing/testsdk/providerserver/providerserver.go

```diff
@@ -57,6 +58,10 @@ type ProviderServer struct {
 	Provider provider.Provider
 }
 
+func (s ProviderServer) GetMetadata(ctx context.Context, request *tfprotov6.GetMetadataRequest) (*tfprotov6.GetMetadataResponse, error) {
+	return &tfprotov6.GetMetadataResponse{}, nil
```

Comment:
Since its relatively straightforward, I'd recommend to go ahead an implement the response correctly similar to the `GetProviderSchema` logic:

```go
resp := &tfprotov6.GetMetadataResponse{
	ServerCapabilities: &tfprotov6.ServerCapabilities{
		GetProviderSchemaOptional: true,
		PlanDestroy: true,
	},
}

for typeName := range s.Provider.DataSourcesMap() {
	resp.DataSources = append(resp.DataSources, tfprotov6.DataSourceMetadata{
		TypeName: typeName,
	}),
}

for typeName := range s.Provider.ResourcesMap() {
	resp.Resources = append(resp.Resources, tfprotov6.ResourceMetadata{
		TypeName: typeName,
	}),
}

return resp, nil
```

Reply:
Thanks for the feedback. Have added your suggested changes.